applications/convection_diffusion_application/custom_elements/laplacian_element.cpp

```diff
@@ -76,14 +76,23 @@ void LaplacianElement::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, Vec
     //reading integration points and local gradients
     const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints();
     const GeometryType::ShapeFunctionsGradientsType& DN_De = GetGeometry().ShapeFunctionsLocalGradients();
-    
+    const Matrix& N_local = GetGeometry().ShapeFunctionsValues();
+
     Element::GeometryType::JacobiansType J0;
     Matrix DN_DX(number_of_points,dim);
     Matrix InvJ0(dim,dim);
     Vector temp(number_of_points);
+    
+    Vector heat_flux_local(number_of_points);
+    for(unsigned int node_element = 0; node_element<number_of_points; node_element++)
+    {
+        heat_flux_local[node_element] = GetGeometry()[node_element].GetSolutionStepValue(HEAT_FLUX) * N_local(0,node_element);
+    }
```

Comment:
i would change this to

    Vector heat_flux_local(number_of_points);
    for(unsigned int node_element = 0; node_element<number_of_points; node_element++)
    {
        heat_flux_local[node_element] = GetGeometry()[node_element].FastGetSolutionStepValue(HEAT_FLUX);
    }

which is 2 changes:
1 - use the Fast version of GetSolutionStepValue
2 - do not use N_local(0,...), it is plain wrong here...